### PR TITLE
feat (client): turn classic show into a functional component

### PR DESCRIPTION
**Pull request by ahmaxed**
- feat: turn classic show into a functional component
- feat: remove getChallenge
- feat: use challenge files and tests from query

Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #XXXXX

### CODE REVIEW

**Comment by ojeytonwilliams** [comment]
Hey @ahmadabdolsaheb, one point of concern is that this creates two sources of truth for `challengeFiles`.  Previously the data Gatsby fetches was only used once, in a call to `createFiles`, and then everything that needed access talked to redux. But now Show also passes Gatsby's `challengeFiles` to its components.

I can't see any obvious bugs resulting from this, but it seems likely that some will arise (or have already, but I haven't seen them).

**Comment by ahmadabdolsaheb** [comment]
@ojeytonwilliams, that makes sense. There are some props that are being passed to the children components directly from the graphql query, but that should be alright. 
I am removing the tests that are being pulled from redux since they don't seem to be used anywhere.

**Review by huyenltnguyen — COMMENTED** [review]
I always love seeing code refactoring 😄 👏

**Comment by ahmadabdolsaheb** [comment]
> I always love seeing code refactoring 😄 👏

I have good news for you  😄   #49690

File: client/src/templates/Challenges/classic/show.tsx [hunk 1]
```diff
@@ -453,126 +407,102 @@ class ShowClassic extends Component<ShowClassicProps, ShowClassicState> {
...
+            hasEditableBoundaries={hasEditableBoundaries}
+            hasNotes={!!notes}
+            hasPreview={hasPreview()}
+            instructions={renderInstructionsPanel({
+              showToolPanel: false
+            })}
+            notes={renderNotes(notes)}
+            preview={renderPreview()}
+            testOutput={renderTestOutput()}
+            // eslint-disable-next-line @typescript-eslint/unbound-method
```
[the remark this answers is not in the document]
Comment (ahmadabdolsaheb) [reply client/src/templates/Challenges/classic/show.tsx:459]: That makes sense.

File: client/src/templates/Challenges/classic/show.tsx [hunk 2]
```diff
@@ -160,41 +157,79 @@ const handleContentWidgetEvents = (e: MouseEvent | TouchEvent): void => {
...
+  createFiles,
+  cancelTests,
+  challengeMounted,
+  initConsole,
+  initTests,
+  updateChallengeMeta,
+  openModal,
+  setIsAdvancing,
+  savedChallenges,
+  t,
```
[the remark this answers is not in the document]
Comment (ahmadabdolsaheb) [reply client/src/templates/Challenges/classic/show.tsx:199]: the less the better

File: client/src/templates/Challenges/classic/show.tsx [hunk 3]
```diff
@@ -160,41 +157,79 @@ const handleContentWidgetEvents = (e: MouseEvent | TouchEvent): void => {
...
-    this.setState({ usingKeyboardInTablist });
-  }
+    // Forcing a state update with the value of each panel since on stop resize
+    // is executed per each panel.
+    if (typeof layout === 'object')
+      setLayout({
+        ...layout,
+        [name]: { flex }
+      });
+    setResizing(false);
```
[the remark this answers is not in the document]
Comment (ahmadabdolsaheb) [reply client/src/templates/Challenges/classic/show.tsx]: Thanks for the suggestion.

[the remark this answers is not in the document]
Comment (Sboonny) [reply client/src/templates/Challenges/classic/show.tsx]: I am surprised that it was returning something and now it doesn't, but it still works. 👍

**Review by huyenltnguyen — APPROVED** [review]
LGTM 👍

**Review by raisedadead — APPROVED** [review]

**Comment by Sboonny** [comment]
Second sorting `editorRef` type

File: client/src/templates/Challenges/classic/show.tsx [hunk 4]
```diff
@@ -210,109 +245,69 @@ class ShowClassic extends Component<ShowClassicProps, ShowClassicState> {
...
+  // layout: Holds the information of the panes sizes for desktop view
+  const [layout, setLayout] = useState(getLayoutState());
+  const [resizing, setResizing] = useState(false);
+  const [usingKeyboardInTablist, setUsingKeyboardInTablist] = useState(false);
+
+  const containerRef = useRef<HTMLElement>();
+  const editorRef =
+    useRef<editor.IStandaloneCodeEditor>() as MutableRefObject<editor.IStandaloneCodeEditor>;
+  const instructionsPanelRef =
+    useRef<HTMLDivElement>() as RefObject<HTMLDivElement>;
```
Comment (Sboonny) [inline client/src/templates/Challenges/classic/show.tsx:259]:
```suggestion
  const editorRef = useRef<editor.IStandaloneCodeEditor>();
  const instructionsPanelRef = useRef<HTMLDivElement>(null);
```

File: client/src/templates/Challenges/classic/show.tsx [hunk 5]
```diff
@@ -453,134 +407,106 @@ class ShowClassic extends Component<ShowClassicProps, ShowClassicState> {
...
+  return (
+    <Hotkeys
+      challengeType={challengeType}
+      editorRef={editorRef}
+      executeChallenge={executeChallenge}
+      innerRef={containerRef}
+      instructionsPanelRef={instructionsPanelRef}
+      nextChallengePath={nextChallengePath}
+      prevChallengePath={prevChallengePath}
+      usesMultifileEditor={usesMultifileEditor}
```
Comment (Sboonny) [inline client/src/templates/Challenges/classic/show.tsx:438]:
```suggestion
      executeChallenge={executeChallenge}
      innerRef={containerRef}
      instructionsPanelRef={instructionsPanelRef}
      nextChallengePath={nextChallengePath}
      prevChallengePath={prevChallengePath}
      usesMultifileEditor={usesMultifileEditor}
      {...(editorRef && { editorRef: editorRef })}
```

File: client/src/templates/Challenges/classic/show.tsx [hunk 6]
```diff
@@ -1,7 +1,13 @@
 import { graphql } from 'gatsby';
-import React, { Component, MutableRefObject } from 'react';
+import React, {
+  useState,
+  useEffect,
+  MutableRefObject,
+  RefObject,
+  useRef
+} from 'react';
```
Comment (Sboonny) [inline client/src/templates/Challenges/classic/show.tsx:8]:
```suggestion
import React, { useState, useEffect, useRef } from 'react';
```

File: client/src/templates/Challenges/components/hotkeys.tsx [hunk 7]
```diff
@@ -61,10 +62,10 @@ interface HotkeysProps {
   challengeFiles: ChallengeFiles;
   challengeType?: number;
   children: React.ReactElement;
-  editorRef?: React.RefObject<HTMLElement>;
+  editorRef?: MutableRefObject<editor.IStandaloneCodeEditor>;
```
Comment (Sboonny) [inline client/src/templates/Challenges/components/hotkeys.tsx:65]:
```suggestion
  editorRef: MutableRefObject<editor.IStandaloneCodeEditor | undefined>;
```

File: client/src/templates/Challenges/classic/editor.tsx [hunk 8]
```diff
@@ -79,7 +79,7 @@ interface EditorProps {
   canFocus: boolean;
   challengeFiles: ChallengeFiles;
   challengeType: number;
-  containerRef: RefObject<HTMLElement>;
+  containerRef: RefObject<HTMLElement | undefined>;
```
Comment (Sboonny) [inline client/src/templates/Challenges/classic/editor.tsx:82]: There is `RefObject` import that can be deleted as well, and `editorRef` type below dimensions that should be updated to this `editorRef: MutableRefObject<editor.IStandaloneCodeEditor| undefined>;`

```suggestion
  containerRef: MutableRefObject<HTMLElement | undefined>;
```

**Review by Sboonny — COMMENTED** [review]
This should the remove the need for `as`

**Pull request merged**